rbc/tests/test_omnisci_caller.py

```diff
@@ -0,0 +1,192 @@
+    assert_equal(arange(3, np.int32(1))['x'], np.arange(3, dtype=np.int32) + 1)
+
+
+def test_remote_float64_evaluation(omnisci):
+    myincr = omnisci.get_caller('myincr')
+    arange = omnisci.get_caller('arange')
+
+    pytest.xfail('SELECT downcasts float64 to float32')
+    assert_equal(myincr(np.float64(3.5)), np.float64(4.5))
+    assert_equal(arange(3, np.float64(1))['x'], np.arange(3, dtype=np.float64) + 1)
+
+
+def test_remote_composite_udf_evaluation(omnisci):
+    myincr = omnisci.get_caller('myincr')
+
+    assert_equal(str(myincr(myincr(3))),
+                 'myincr(CAST(myincr(CAST(3 AS BIGINT)) AS BIGINT))')
+    assert_equal(str(myincr(myincr(3, hold=False))), 'myincr(CAST(4 AS BIGINT))')
+    assert_equal(myincr(myincr(3), hold=False), 5)
+    assert_equal(myincr(myincr(3)).execute(), 5)
+
+
+def test_remote_udtf_evaluation(omnisci):
+    arange = omnisci.get_caller('arange')
+
+    assert_equal(str(arange(3, 1)),
+                 'SELECT x FROM TABLE(arange(CAST(3 AS INT), CAST(1 AS BIGINT)))')
+
+    assert_equal(arange(3, 1)['x'], list(np.arange(3, dtype=np.int64) + 1))
+    assert_equal(arange(3, 1.5)['x'], list(np.arange(3, dtype=np.float64) + 1.5))
+    assert_equal(arange(np.int32(3), 1)['x'], list(np.arange(3, dtype=np.int32) + 1))
+    assert_equal(arange(3, np.float32(1))['x'], np.arange(3, dtype=np.float32) + 1)
+    assert_equal(arange(3, np.int64(1))['x'], np.arange(3, dtype=np.int64) + 1)
+
+
+def test_remote_composite_udtf_evaluation(omnisci):
+    arange = omnisci.get_caller('arange')
+    aincr = omnisci.get_caller('aincr')
+    myincr = omnisci.get_caller('myincr')
+
+    r = aincr(arange(3, 1), 2)
+    assert_equal(str(r), 'SELECT y FROM TABLE(aincr(CURSOR(SELECT x FROM'
+                 ' TABLE(arange(CAST(3 AS INT), CAST(1 AS BIGINT)))), CAST(2 AS BIGINT)))')
+
+    r = r.execute()
+    assert_equal(r['y'], np.arange(3, dtype=np.int64) + 1 + 2)
+
+    r = arange(3, myincr(2, hold=False))
+    assert_equal(str(r), 'SELECT x FROM TABLE(arange(CAST(3 AS INT), CAST(3 AS BIGINT)))')
+    assert_equal(r['x'], np.arange(3, dtype=np.int64) + 2 + 1)
+
+
+def test_remote_composite_udtf_udf(omnisci):
+    """
+    TableFunctionExecutionContext.cpp:277 Check failed:
+    col_buf_ptrs.size() == exe_unit.input_exprs.size() (1 == 2)
+    """
+    myincr = omnisci.get_caller('myincr')
+    arange = omnisci.get_caller('arange')
+
+    r = arange(3, myincr(2))
+    assert_equal(str(r), ('SELECT x FROM TABLE(arange(CAST(3 AS INT),'
+                          ' CAST(myincr(CAST(2 AS BIGINT)) AS BIGINT)))'))
+
+    pytest.xfail('udtf(udf) crashes omniscidb server')
+    assert_equal(r['x'], np.arange(3, dtype=np.int64) + 2 + 1)
+
+
+def test_remote_udf_typeerror(omnisci):
+    myincr = omnisci.get_caller('myincr')
+    try:
+        myincr("abc")
+    except TypeError as msg:
+        assert_equal(str(msg), '''\
+found no matching function type to given argument types:
+    string
+  available function types:
```

Comment:
Noting this here. Not sure about `function types`, maybe we should talk about signature instead. Also it should say that it's the input type which is causing a problem. Because right now some users can be confused and not get why there is just one type and below there is a list of signatures

Reply:
@tupui Are you suggesting the replacement: `function type` -> `signature type`? Or changing the whole exception message? Could you provide the desired message sample?

Reply:
Personally, I read the following better
```
found no matching function signature for given argument types:
     (string) -> ??
   available function signatures:
     (int32) -> int32
     (int64) -> int64
     (float32) -> float32
     (float64) -> float64
```

Reply:
Fixed. Using `(string) -> ...` though as question marks have negative energy ;)

Reply:
haha, perfect thanks 👍